presto-pinot/src/main/java/io/prestosql/pinot/PinotConnectorFactory.java

```diff
@@ -23,12 +25,20 @@
 import org.weakref.jmx.guice.MBeanModule;
 
 import java.util.Map;
+import java.util.Optional;
 
 import static java.util.Objects.requireNonNull;
 
 public class PinotConnectorFactory
         implements ConnectorFactory
 {
+    private final Optional<Module> extension;
+
+    public PinotConnectorFactory(Optional<Module> extension)
```

Comment:
Is this extension needed for host mapper? If not please use separate commit.

Reply:
Yes, it's needed for the hostmapper.